llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -1270,6 +1336,30 @@ void llvm::AIEPostRASchedStrategy::buildGraph(ScheduleDAGMI &DAG, AAResults *AA,
   static_cast<AIEScheduleDAGMI &>(DAG).recordDbgInstrs(Region);
 }
 
+void AIEPostRASchedStrategy::addFixedSUnit(MachineInstr &MI, bool IsTop,
+                                           unsigned Dist) {
+  assert(!(IsTop && FirstBotFixedSU) && "Top-fixed SUnits must be added first");
+  assert(DAG->SUnits.size() < DAG->SUnits.capacity() &&
+         "SUnits need to be re-allocated.");
+  unsigned SUNum = DAG->initSUnit(MI).value();
```

Comment:
nit: ignoring has_value() here.

Reply:
That would be an error if the value is missing

Reply:
and we don't like asserts? :-b

Reply:
I mean that I would get to the same behaviour: the error would be caught and not silently ignored.